src/dashboard/utilities.py

```diff
@@ -46,3 +47,72 @@ def toggle_classname(class_str: str, class_to_toggle: str) -> str:
         return set_classname(class_str, class_to_toggle, False)
     else:
         return set_classname(class_str, class_to_toggle, True)
+
+
+def pluralize(unit: str, amount: int) -> str:
+    """Pluralize a unit if the amount is greater than 1.
+
+    Args:
+        unit (str): The unit.
+        amount (int): The amount.
+
+    Returns:
+        str: The unit with s appended
+        if amount is greater than 1,
+        otherwise the unit.
+    """
+    if amount == 1:
+        return unit
+
+    return f"{unit}s"
+
+
+def singularize(article: str, amount: int) -> str:
+    """Singularize an amount if it is equal to one.
+
+    Args:
+        article (str): The indefinite
+        article of an amount of something.
+        E.g. "A" or "An" in english.
+        amount (int): An amount.
+
+    Returns:
+        str: `article` if the amount
+        is equal to 1, otherwise
+        a string of amount.
+    """
+    if amount == 1:
+        return article
+
+    return str(amount)
+
+
+def to_human_time_delta(t1: datetime, t2: datetime) -> str:
+    """Convert a time delta to a human readable form.
+
+    Args:
+        t1 (datetime): A point in time.
+        t2 (datetime): A point in time.
+
+    Returns:
+        str: A human readable form of
+        the delta.
+    """
+    delta = abs(t1 - t2)
+    units: list[tuple[str, str, int]] = [
+        ("A", "week", 604800),
+        ("A", "day", 86400),
+        ("An", "hour", 3600),
+        ("A", "minute", 60),
+        ("A", "second", 1),
+    ]
+    for article, unit, duration_seconds in units:
+        unit_duration = int(delta.total_seconds()) // duration_seconds
+        if unit_duration >= 1:
+            return (
+                f"{singularize(article, unit_duration)} {pluralize(unit, unit_duration)} ago"
+                if unit != "week"
+                else min(t1, t2).strftime("%d %b. %Y")
```

Comment:
I'm not sure showing the date instead of something like "a month ago" is preferable? Most websites (YouTube, Reddit) seem to always use units (i.e month, years). Perhaps we could show exact date when hovering over the duration text?